sdk/go/common/util/gitutil/git.go

```diff
@@ -857,6 +862,52 @@ func GitListBranchesAndTags(url string) ([]plumbing.ReferenceName, error) {
 	return results, nil
 }
 
+// GetLatestTagOrHash returns the latest tag or hash in the repository.
+// To do this, we list all the tags in the repository and try to pars them as semver.  If we can't
+// find any valid tags that are semver, we'll use the hash of the default branch.  If we can't find
+// any tags or default branches we return an error.
+func GetLatestTagOrHash(ctx context.Context, url string) (*semver.Version, error) {
+	refs, err := gitListRefs(ctx, url)
+	if err != nil {
+		return nil, err
+	}
+
+	var versions []semver.Version
```

Comment:
Nit: you don't really need to collect all versions, you could instead track a single "max" version candidate and avoid sorting the array at the end.